### PR TITLE
[sdk-logs] Expose logging SDK implementation as public

### PR DESCRIPTION
**Pull request by CodeBlanch**
Relates to #4433

## Changes

* Exposes the logging implementation in `OpenTelemetry` and removes the added `InternalsVisibleTo`s

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [X] Unit tests added/updated
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry/Logs/LogRecord.cs [hunk 1]
```diff
@@ -157,7 +160,7 @@ public ActivityTraceFlags TraceFlags
     /// <summary>
     /// Gets or sets the log <see cref="Microsoft.Extensions.Logging.LogLevel"/>.
     /// </summary>
-    // todo: [Obsolete("Use Severity instead LogLevel will be removed in a future version.")]
+    [Obsolete("Use Severity instead LogLevel will be removed in a future version.")]
```
Comment (utpilla) [inline src/OpenTelemetry/Logs/LogRecord.cs:163]: nit:
```suggestion
    [Obsolete("Use Severity instead. LogLevel will be removed in a future version.")]
```

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry/Logs/LogRecord.cs (hunk 1 above)
Comment (cijothomas) [inline src/OpenTelemetry/Logs/LogRecord.cs:163]:
```suggestion
    [Obsolete("Use Severity instead LogLevel will be removed in a future release with a major version.")]
```

**Review by vishweshbankwar — APPROVED** [review]

File: src/OpenTelemetry/Logs/LogRecord.cs (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/Logs/LogRecord.cs]: Updated

File: src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs [hunk 2]
```diff
@@ -74,7 +74,15 @@ public override ExportResult Export(in Batch<LogRecord> batch)
                     this.WriteLine($"{"LogRecord.CategoryName:",-RightPaddingLength}{logRecord.CategoryName}");
                 }
 
-                this.WriteLine($"{"LogRecord.LogLevel:",-RightPaddingLength}{logRecord.LogLevel}");
+                if (logRecord.Severity.HasValue)
```
[the remark this answers is not in the document]
Comment (CodeBlanch) [reply src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs:77]: I added entries. LMK if you are good with the verbiage.

File: src/OpenTelemetry/Logs/LogRecord.cs (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/Logs/LogRecord.cs]: I think a bit more discussion is needed around our strategy on this and then we can land a policy doc + update all the warnings we have sprinkled around?

/cc @alanwest

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,22 @@
...
 
+* Added [Logs Bridge
+  API](https://github.com/open-telemetry/opentelemetry-specification/blob/976432b74c565e8a84af3570e9b82cb95e1d844c/specification/logs/bridge-api.md)
+  implementation (`Sdk.CreateLoggerProviderBuilder`, etc.).
+  ([#4433](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4433))
+
+* Obsoleted `LogRecord.LogLevel` in favor of the `LogRecord.Severity` property
+  which matches the [OpenTelemetry Specification > Logs DataModel > Severity
+  definition](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-severitynumber).
+  ([#4433](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4433))
```
Comment (utpilla) [inline src/OpenTelemetry/CHANGELOG.md:17]: The PR number needs to be corrected.

**Pull request merged**